### PR TITLE
Fix various bugs with student/cohort assignment

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes ilios/ilios#5390
fixes ilios/ilios#5391
fixes ilios/ilios#5392
fixes ilios/ilios#5396

todo: 
- [x] update/add test coverage

### CODE REVIEW

**Review by dartajax — APPROVED** [review]
I added a cpl annoying and cloying comments in two of the issues wondering whether some of the previously selected information can be retained if a user switches back to the previous school or undoes a filter operation. I approve it this way and am ready to support and document this behavior. Feel free to veto my comments. I approve of this the way it is - good catches and fixes.

File: packages/frontend/app/templates/assign-students.hbs [hunk 1]
```diff
@@ -1,36 +1,9 @@
...
-  <AssignStudents
-    @school={{this.selectedSchool}}
-    @students={{this.filteredUnassignedStudents}}
-  />
-</section>
+<AssignStudents::Root
+  @model={{this.model}}
+  @query={{this.query}}
+  @schoolId={{this.schoolId}}
+  @setSchoolId={{this.setSchoolId}}
```
Comment (jrjohnson) [inline packages/frontend/app/templates/assign-students.hbs:7]: I'm fine with this as is, but I also wouldn't mind ditching the actions in the controller and using `@setSchoolId={{set this.schoolId}}` here and below on set query.

File: packages/frontend/translations/en-us.yaml [hunk 2]
```diff
@@ -247,6 +247,7 @@ general:
   newUsersCreatedWarning: "Some users were created, but there were errors."
   noAccountExists: "Your account {accountName} does not match any user records in Ilios. If you need further assistance, please contact your school’s Ilios administrator."
   noAvailableCohorts: No available cohorts
+  noCohortsAvailableForStudentAssignment: There are no cohorts available for student assignment in this school.
```
Comment (jrjohnson) [inline packages/frontend/translations/en-us.yaml:250]: Nice UX add!

**Review by jrjohnson — APPROVED** [review]
Great refactor, top level component pattern is much better and tests are excellent. I had one suggestion, take it or leave it.

File: packages/frontend/app/templates/assign-students.hbs (hunk 1 above)
Comment (stopfstedt) [reply packages/frontend/app/templates/assign-students.hbs:7]: good point. i changed it. :heavy_check_mark:

**Pull request merged**